test/test_autograd.py

```diff
@@ -2705,6 +2705,52 @@ def test_profiler(self):
                 last_end = info.cpu_interval.end
             self.assertEqual(info.name, expected_name)
 
+    def test_profiler_seq_nr(self):
+        with profile() as p:
+            x = torch.randn(10, 10, requires_grad=True)
+            y = torch.randn(10, 10, requires_grad=True)
+            z = x + y
+            s = z.sum()
+            s.backward()
+        # expecting aten::add, aten::sum to have the sequence numbers,
+        # expecting the corresponding backward nodes to have the same numbers
+        # as the forward ops
+        add_seq_nr = -1
+        sum_seq_nr = -1
+        found_add = found_sum = False
+        found_bwd_add = found_bwd_sum = False
+        found_empty = False
+        for e in p.function_events:
```

Comment:
I assume this is after filtering, so you won't have multiple function events with the same name?

Reply:
yes, this issue with double logging is not visible here, because its after filtering;
I also err'd on the side of failing a test, so added extra checks.